File: src/Linked List/Append A Node.cpp
#include <bits/stdc++.h>
using namespace std;

typedef struct node Node;

struct node {
    int data;
    Node *next;
};

Node *head = NULL;

Node *create_node (int item, Node *next)
{
    Node *new_node = (Node *) malloc(sizeof(Node));
    if (new_node == NULL)
    {
        printf("Error! Couldn't create a new node\n");
        exit(1);
    }
    new_node->data = item;
    new_node->next = next;
}

Node *append(Node *head, int item)
{
    Node *new_node = create_node(item, NULL);
    if(head == NULL){
        return new_node;
    }
    Node *current_node = head;
    while (current_node->next != NULL){
        current_node = current_node->next;
    }
    current_node->next = new_node;
    return head;
}

int main()
{
    Node *a = (Node *) malloc(sizeof(Node));
    Node *b = (Node *) malloc(sizeof(Node));
    Node *c = (Node *) malloc(sizeof(Node));

    a->data=10;
    a->next=b;
    b->data=20;
    b->next=NULL;
    //Appending 30 after A then B
    head = a;
    append(head, 30);
    //Printing linked list
    Node *current_node = head;
    while(current_node!=NULL)
    {
        printf("%d\n",current_node->data);
        current_node=current_node->next;
    }
    return 0;
}

File: src/Data_Structure/Linked_List/Append_A_Node.java
package Data_Structure.Linked_List;

public class Append_A_Node {

  static Node head = null;
  static Node createNode (int item, Node next) {
    Node newNode = new Node(item);
    newNode.next = next;
    return newNode;
  }

  static Node append(Node head, int item) {
    Node newNode = new Node(item);

    if (head == null) {
      return newNode;
    }

    Node currentNode = head;
    while (currentNode.next != null) {
      currentNode = currentNode.next;
    }

    currentNode.next = newNode;

    return head;
  }

  public static void main(String[] args) {
    Node a = new Node(10);
    Node b = new Node(20);

    a.next = b;


    head = a;
    append(a, 30);

    Node currentNode = head;
    while (currentNode != null) {
      System.out.println(currentNode.data);
      currentNode = currentNode.next;
    }
  }
}
